graph/pull.go

```diff
@@ -457,6 +463,9 @@ func (s *TagStore) pullV2Tag(r *registry.Session, out io.Writer, endpoint *regis
 		}
 		downloads[i].img = img
 
+		layers = append(layers, img.ID)
+		s.graph.Retain([]string{img.ID}, manifest.Name)
```

Comment:
I made a gist with a trivial example of the problem: https://gist.github.com/jlhawn/1ef3d78d1cb1ac6b666b